### DISCUSSION TITLE
How to set session variables if using Auth0

### DISCUSSION THREAD

**Discussion by achen187** [hasura/graphql-engine#7246] [Question] [answered]
My users have different states, and I only want them to have select access to their states. I want to add a session variable like:

X-Hasura-Allowed-States: ["NY", "CA", "WA"]

And then in my permission expression:

{
  "state": {
    "_in": "X-Hasura-Allowed-States"
  }
}

I'm using Auth0 as my authentication service. I can't seem to find any tutorials on how to set X-Hasura-Allowed-States per user.

**Comment by praveenweb** [comment, marked as answer]
Auth0 has rules which lets you configure the custom claims for JWT. Here's one [example](https://hasura.io/learn/graphql/hasura/authentication/2-custom-jwt-claims-rule/) of how to set it up. You can add in any number of custom `x-hasura-*` session variables inside that.
